### ISSUE TITLE
Can't aggregate nested field in bar chart.

### ISSUE THREAD

**Issue by iliatimofeev** [vega/vega-lite#3744]
Can't aggregate nested field in bar chart. Possible the same as  #3727

```json
{
  "mark": "bar",
  "encoding": {
    "y": { "field": "properties.variety", "type": "nominal"},
    "x": {
      "aggregate": "sum",
      "field": "properties.yield",
      "type": "quantitative"
    },
    "color": {"title": "site", "field": "properties.site", "type": "nominal"}
  },
  "data": {
    "values": {
      "features": [
        {
          "properties": {
            "variety": "Manchuria",
            "yield": 27,
            "site": "University Farm",
            "year": 1931
          }
        },
       {
          "properties": {
            "variety": "Wisconsin No. 38",
            "yield": 29.33333,
            "site": "Duluth",
            "year": 1932
          }
        }
      ],
      "type": "FeatureCollection"
    },
    "format": {"type": "json", "property": "features"}
  },
  "$schema": "https://vega.github.io/schema/vega-lite/v2.4.1.json"
}
```
![export 2](https://user-images.githubusercontent.com/11561581/40025317-e4f1b4f4-57d9-11e8-90cc-49d5b3ef81f3.png)
live demo https://iliatimofeev.github.io/sharer/testreport/#horizontal_stacked_bar_chart

**Comment by domoritz** [comment]
Found the issue and fixed it in #4021.

**Comment by domoritz** [comment]
![image](https://user-images.githubusercontent.com/589034/42730145-6e6e67a2-87a2-11e8-9def-bc1fd4302b71.png)
![image](https://user-images.githubusercontent.com/589034/42730147-77d122bc-87a2-11e8-8ceb-a021accbd686.png)

@iliatimofeev Thank you so much for the perfect examples to reproduce the issue! This made it so much easier for me to fix this.
